src/cache_memoize/__init__.py

```diff
@@ -102,6 +100,7 @@ def _make_cache_key(*args, **kwargs):
 
         @wraps(func)
         def inner(*args, **kwargs):
+            cache = caches[cache_alias]
```

Comment:
Both this line and the line in `invalidate()` now needs a code comment. 
One year from now, someone is going to see that code and wonder "Why isn't this set up once in the decorator context above?! Surely that's awesomer" And they (e.g. me) will fail to remember to trace the git blame back to this PR's wonderful explanation.

Reply:
I've added a brief comment (compared in to the PR wall of text :) ), hopefully this and the test will be enough!

Reply:
The commit message should also have the same description, so hopefully this will be ok.